src/tracked_struct.rs

```diff
@@ -151,10 +151,15 @@ where
 /// This is the key to a hashmap that is (initially)
 /// stored in the [`ActiveQuery`](`crate::active_query::ActiveQuery`)
 /// struct and later moved to the [`Memo`](`crate::function::memo::Memo`).
-#[derive(Debug, PartialEq, Eq, Hash, PartialOrd, Ord, Copy, Clone)]
+#[derive(Debug, PartialEq, Eq, PartialOrd, Ord, Copy, Clone)]
 pub(crate) struct Identity {
-    /// Hash of fields with id attribute
-    identity_hash: IdentityHash,
+    // conceptually, this contains an `IdentityHash`, but using the type directly will grow the size
+    // of if this struct by a word due to unusable padding, so we store the fields directly instead.
+    /// Index of the tracked struct ingredient.
```

Comment:
two nits:
- Phrasing
- Considering moving this comment up to `Identity`'s doc comment?

```suggestion
    /// Index of the tracked struct ingredient.
    ///
    /// Conceptually, this contains an `IdentityHash`, but using `IdentityHash` directly will grow the size
    /// of this struct struct by a `std::mem::size_of::<usize>()` due to unusable padding. To avoid this increase
    /// in size, `Identity` stores the fields directly instead.
```

Reply:
I put it outside of the doc comment as this is just an implementation detail, it is not relevant to the `ingredient_index` field